wp-content/themes/maxbstheme/inc/Tender_Metaboxes.php

```diff
@@ -0,0 +1,209 @@
+                $select,
+                $option['value'],
+                $option['label']
+            );
+        };
+        printf('</select><br/><span class="description">%s</span>', $field['desc']);
+    }
+
+    // Markup for input "DATETIME-LOCAL" and "NUMBERS"
+    private function datetime_and_number_markup($field)
+    {
+        $current_value = get_post_meta($_GET['post'], $field['id'], true);
+        printf('<input type="%s" id="%s" name="%s" value="%s"><br/><span class="description">%s</span>',
+            $field['type'],
+            $field['id'],
+            $field['id'],
+            $current_value,
+            $field['desc']
+        );
+    }
+
+    // Markup for rates table
+    private function table_markup($field)
+    {
+        global $post;
+        $tender_rate_db = new Tender_Rates_DB();
+
+        $rating = $tender_rate_db->get_rating($post->ID);
+
+        printf('<span class="description">%s</span><table>', $field['desc']);
+        foreach ($rating as $rate) {
+            printf(
+                '<tr><td>%s</td><td>%d грн.</td><td>%s</td></tr>',
+                $rate['display_name'],
+                $rate['price'],
+                $rate['rate_time']
+            );
+        }
+        echo '</table>';
+    }
+
+    // Rendering of fields
+    private function fields_markup($field)
+    {
+        switch ($field['type']) {
+            case 'select':
+                $this->select_markup($field);
+                break;
+            case 'number':
+            case 'datetime-local':
+                $this->datetime_and_number_markup($field);
+                break;
+            case 'table':
+                $this->table_markup($field);
+        }
+    }
+
+    // Show fields in form
+    private function show_fields($fields)
+    {
+        echo '<table class="form-table">';
+        foreach ($fields as $field) {
+            printf('<tr><th><label for="%s">%s</label></th><td>',
+                $field['id'],
+                $field['label']
+            );
+            $this->fields_markup($field);
+            echo '</td></tr>';
+        }
+        echo '</table>';
+    }
+
+    public function show_tender_metabox() {
+        $tender_meta_fields = $this->tender_meta_fields();
+
+        echo '<input type="hidden" name="custom_meta_box_nonce" value="' .
+            wp_create_nonce(basename(__FILE__)) .
+            '" />';
+
+        $this->show_fields($tender_meta_fields);
+    }
+
+    private function write_meta_to_db($fields, $post_id)
+    {
+        foreach ($fields as $field) {
+            $old = get_post_meta($post_id, $field['id'], true);
+
+            $new = $_POST[$field['id']];
+            if($new && $new != $old) {
+                update_post_meta($post_id, $field['id'], $new);
+            } elseif ('' == $new && $old) {
+                delete_post_meta($post_id, $field['id'], $old);
+            }
+        }
+    }
+
+    // Check values from metaboxes and write to database
```

Comment:
/**
	 * Check values from metaboxes and write to database
	 * 
	 * @param $post_id
	 * @return mixed
	 */